examples/plotting/file/clustering.py

```diff
@@ -0,0 +1,64 @@
+""" Example inspired by an example from the scikit-learn project:
+http://scikit-learn.org/stable/auto_examples/cluster/plot_cluster_comparison.html
+"""
+
+import time
+
+import numpy as np
+
+from sklearn import cluster, datasets
+from sklearn.neighbors import kneighbors_graph
+from sklearn.preprocessing import StandardScaler
+
+from bokeh.plotting import figure, show, output_file
+from bokeh.models import HBox, VBox
+
+N = 50000
+PLOT_SIZE = 400
+
+# Generate datasets. 
+np.random.seed(0)
+noisy_circles = datasets.make_circles(n_samples=N, factor=.5, noise=.04)
+noisy_moons = datasets.make_moons(n_samples=N, noise=.05)
+centers = [(-2, 3), (2, 3), (-2, -3), (2, -3)]
+blobs1 = datasets.make_blobs(centers=centers, n_samples=N, cluster_std=0.4, random_state=8)
+blobs2 = datasets.make_blobs(centers=centers, n_samples=N, cluster_std=0.7, random_state=8)
+
+colors = np.array([x for x in ('#00f', '#0f0', '#f00', '#0ff', '#f0f', '#ff0')])
+colors = np.hstack([colors] * 20)
+
+# Create clustering algorithms
+dbscan = cluster.DBSCAN(eps=.2)
+birch = cluster.Birch(n_clusters=2)
+two_means = cluster.MiniBatchKMeans(n_clusters=2)
+spectral = cluster.SpectralClustering(n_clusters=2,
+                                        eigen_solver='arpack',
+                                        affinity="nearest_neighbors")
+affinity_propagation = cluster.AffinityPropagation(damping=.9,
+                                                    preference=-200)
```

Comment:
I don't always think this sort of thing is worthwhile, but I do think this is nicer for an example:
```
dbscan               = cluster.DBSCAN(eps=.2)
birch                = cluster.Birch(n_clusters=2)
two_means            = cluster.MiniBatchKMeans(n_clusters=2)
affinity_propagation = cluster.AffinityPropagation(damping=.9, preference=-200)
spectral             = cluster.SpectralClustering(n_clusters=2, 
                                                  eigen_solver='arpack', 
                                                  affinity="nearest_neighbors")
```